commands/equip_commands.py

```diff
@@ -0,0 +1,258 @@
+#
+# Commands and cmdsetRelated to the equip handler
+#
+from evennia import CmdSet, utils
+from commands.command import Command
+
+class EquipCmdSet(CmdSet):
+
+    key = "equip_cmdset"
+    priority = 2
+
+    def at_cmdset_creation(self):
+        "Populate CmdSet"
+        self.add(CmdEquip())
+        self.add(CmdWear())
+        self.add(CmdWield())
+        self.add(CmdHold())
+        self.add(CmdRemove())
+
+class CmdEquip(Command):
+    """
+    equip
+
+    Usage:
+      equip
+
+    Show your current equipment.
+
+    """
+    key = "equip"
+    aliases = ["eq"]
+    locks = "cmd:all()"
+
+    def func(self):
+        "check equipment"
+        caller = self.caller
+        table = utils.prettytable.PrettyTable(["slot", "item"])
+        table.header = False
+        table.border = False
+        for slot, item in caller.equip:
+            if not item or not item.access(caller, 'view'):
+                continue
+            table.add_row(["{C%s{n" % slot.capitalize(), item.name.capitalize()])
+        if not str(table):
+            string = "You have nothing in your equipment."
+        else:
+            string = "{wYour equipment:\n%s" % table
+        self.caller.msg(string)
+
+class CmdWear(Command):
+    """
+    wear
+
+    Usage:
+      wear <obj>
+
+    Wear an object.
+
+    """
+    key = "wear"
+    locks = "cmd:all()"
+
+    def func(self):
+        "implements the command."
+
+        caller = self.caller
+        args = self.args.strip()
+
+        if not args:
+            caller.msg("Wear what?")
+            return
+
+        # this will search for a target
+        obj = caller.search(args)
+
+        if not obj:
+            return
+
+        if not obj in caller.contents:
+            caller.msg("You don't have %s in your inventory.")
+
+        # equip primary hand with the proper command
+        if 'right hand' in utils.make_iter(obj.db.slot):
+            caller.execute_cmd("wield %s" % args)
+            return
+
+        if not obj.db.slot or 'holds' in utils.make_iter(obj.db.slot)  \
+           or not obj.access(caller, 'equip'):
+            caller.msg("You can't equip %s." % obj)
+            return
+
+        if obj in caller.equip.items():
+            caller.msg("You already equip %s." % obj.name)
+            return
+
+        if not caller.equip.add(obj):
+            string = "You can't equip %s." % obj
+            if [s for s in utils.make_iter(obj.db.slot) if s in caller.equip.empty_slots()]:
+                string += " You have already something there."
+            caller.msg(string)
+            return
+        # call hook
+        if hasattr(obj, "at_equip"):
+            obj.at_equip(caller)
+        caller.msg("You wear %s." % obj)
+        caller.location.msg_contents("%s wears %s." % (caller.name.capitalize(), obj),
+                                     exclude=caller)
+
+
+class CmdWield(Command):
```

Comment:
Yes I think your proposal has a lot of good points. I'll think about this today but probably we'll keep the duplication.